### PR TITLE
OSDOCS-10637 Adding warnings re. using shared VPCs for HCP clusters

### PR DESCRIPTION
**Pull request by laubai**
Version(s):
4.15+

Issue:
https://issues.redhat.com/browse/OSDOCS-10637

Link to docs preview:
https://76339--ocpdocs-pr.netlify.app/openshift-rosa/latest/rosa_hcp/rosa-hcp-creating-cluster-with-aws-kms-key.html
https://76339--ocpdocs-pr.netlify.app/openshift-rosa/latest/rosa_hcp/rosa-hcp-sts-creating-a-cluster-ext-auth.html
https://76339--ocpdocs-pr.netlify.app/openshift-rosa/latest/rosa_hcp/rosa-hcp-sts-creating-a-cluster-quickly.html
https://76339--ocpdocs-pr.netlify.app/openshift-rosa/latest/rosa_install_access_delete_clusters/rosa-shared-vpc-config.html

QE review:
- [x] QE has approved this change.

Additional information:
Thanks to Paul Foster for the report

### CODE REVIEW

File: snippets/imp-rosa-hcp-no-shared-vpc-support.adoc [hunk 1]
```diff
@@ -0,0 +1,4 @@
+[IMPORTANT]
+====
+Shared VPCs are not supported for use with {hcp-title}. Do not install a {hcp-title} cluster in an environment that uses Amazon VPC sharing.
```
Comment (wgordon17) [inline snippets/imp-rosa-hcp-no-shared-vpc-support.adoc:3]:
```suggestion
link:https://docs.aws.amazon.com/vpc/latest/userguide/vpc-sharing.html[Sharing VPCs across multiple AWS accounts] is not currently supported for {hcp-title}. Do not install a {hcp-title} cluster into subnets shared from another AWS account. See link:https://access.redhat.com/solutions/6980058["Are multiple ROSA clusters in a single VPC supported?"] for more information.
```

File: rosa_install_access_delete_clusters/rosa-shared-vpc-config.adoc [hunk 2]
```diff
@@ -11,6 +11,11 @@ ifdef::openshift-rosa[]
 endif::openshift-rosa[]
 clusters in shared, centrally-managed AWS virtual private clouds (VPCs). 
 
+[IMPORTANT]
+====
+Shared VPC environments are supported only for ROSA Classic clusters using STS for authentication.
```
Comment (wgordon17) [inline rosa_install_access_delete_clusters/rosa-shared-vpc-config.adoc:16]:
```suggestion
link:https://docs.aws.amazon.com/vpc/latest/userguide/vpc-sharing.html[Sharing VPCs across multiple AWS accounts] is currently only supported for ROSA Classic clusters using STS for authentication.
```

File: rosa_hcp/rosa-hcp-creating-cluster-with-aws-kms-key.adoc [hunk 3]
```diff
@@ -16,7 +16,7 @@ Create a {product-title} (ROSA) with a {hcp} (HCP) cluster  using a custom AWS K
 
 To create a {hcp-title} cluster, you must have the following items:
 
-* A configured virtual private cloud (VPC)
+* A configured virtual private cloud (VPC) that is not shared between multiple AWS accounts
```
Comment (wgordon17) [inline rosa_hcp/rosa-hcp-creating-cluster-with-aws-kms-key.adoc:19]: I worry that this additional text may cause unnecessary confusion. I suspect just adding the admonition snippet should be sufficient.

**Review by wgordon17 — CHANGES_REQUESTED** [review]

File: rosa_hcp/rosa-hcp-creating-cluster-with-aws-kms-key.adoc (hunk 3 above)
Comment (laubai) [reply rosa_hcp/rosa-hcp-creating-cluster-with-aws-kms-key.adoc:19]: Is that because it's too wordy? Would it be better to say "that is not a shared VPC"?

Comment (laubai) [reply rosa_hcp/rosa-hcp-creating-cluster-with-aws-kms-key.adoc]: Removed for now, we can add in future if needed.

Comment (wgordon17) [reply rosa_hcp/rosa-hcp-creating-cluster-with-aws-kms-key.adoc]: A "Shared VPC" is honestly a pretty _unusual_ configuration in the first place, so if we add these "weird callouts" by changing headers and what not, I worry it will end up causing _more_ confusion than it helps
